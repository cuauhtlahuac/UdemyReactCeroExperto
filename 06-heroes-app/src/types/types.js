export const types = {
	login: Symbol('login'), 
	logout: Symbol('logout'),
};
