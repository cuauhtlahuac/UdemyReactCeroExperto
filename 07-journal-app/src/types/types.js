export const types = {
	login: 'Auth/login',
	logout: 'Auth/logout',

	uiSetError: 'UI/Set error',
	uiRemoveError: 'UI/Remove error',
	uiStartLoading: 'UI/Start loading',
	uiFinishLoading: 'UI/Finish loading',

	notesAddEntry: 'Notes/New note',
	notesActive: 'Notes/Set active note',
	notesLoad: 'Notes/Load notes',
	notesUpdated: 'Notes/Updated note',
	notesFileUrl: 'Notes/Update file url',
	notesDelete: 'Notes/Delete note',
	notesLogoutAndClean: 'Notes/Logout and clean',
	notesAddNew: 'Notes/Add new note',
};
