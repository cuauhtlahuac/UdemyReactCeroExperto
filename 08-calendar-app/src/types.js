export const uniqueTypes = {
	uiOpenModal: Symbol('ui/open-modal'),
	uiCloseModal: Symbol('ui/close-modal'),

	eventSetActive: Symbol('event/set-active'),
	eventAddNew: Symbol('event/add-new'),
	cleanActiveEvent: Symbol('event/clean-active-event'),
	saveActiveEvent: Symbol('event/save-active-event'),
	deleteEvent: Symbol('event/delete-event'),
};

export const types = {
	[uniqueTypes.uiOpenModal]: 'ui/open-modal',
	[uniqueTypes.uiCloseModal]: 'ui/close-modal',

	[uniqueTypes.eventSetActive]: 'event/set-active',
	[uniqueTypes.eventAddNew]: 'event/add-new',
	[uniqueTypes.cleanActiveEvent]: 'event/clean-active-event',
	[uniqueTypes.saveActiveEvent]: 'event/save-active-event',
	[uniqueTypes.deleteEvent]: 'event/delete-event',
};
