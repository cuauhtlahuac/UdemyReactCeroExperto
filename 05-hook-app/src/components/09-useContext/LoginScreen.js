import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
	const { setUser } = useContext(UserContext);
	return (
		<div>
			<h1>LoginScreen</h1>
			<hr />
			<button
				className="btn btn-primary"
				onClick={() => setUser({ id: 132, name: 'Petronila' })}
			>
				Click
			</button>
		</div>
	);
};

export default LoginScreen;
