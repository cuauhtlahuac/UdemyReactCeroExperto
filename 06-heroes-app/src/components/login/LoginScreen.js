import React, { useContext } from 'react';
import { AuthContext } from 'contexts/authContext';
import { types } from 'types/types';

const LoginScreen = ({ history }) => {
	const { dispatch } = useContext(AuthContext);

	const handleLogin = () => {
		dispatch({
			type: types.login,
			payload: { name: 'Cuauhtlahuac' },
		});

		history.replace('/');
	};

	return (
		<div className="m-5">
			<h1>Login</h1>
			<hr />

			<button className="btn btn-primary" onClick={handleLogin}>
				Login
			</button>
		</div>
	);
};

export default LoginScreen;
