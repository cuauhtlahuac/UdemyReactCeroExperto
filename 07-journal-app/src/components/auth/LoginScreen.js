import React from 'react'

const LoginScreen = () => {
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
    </div>
  )
}

export default LoginScreen
