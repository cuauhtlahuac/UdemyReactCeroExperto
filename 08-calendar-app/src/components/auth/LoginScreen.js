import { useForm } from 'hooks/useForm';

import './login.css';

const LoginScreen = () => {
	const initialLoginState = {
		loginEmail: '',
		loginPassword: '',
	};

	const [ formLoginValues, handleLoginInputChange, reset ] = useForm(
		initialLoginState,
	);

	const { loginEmail, loginPassword } = formLoginValues;

	const handleLogin = event => {
		event.preventDefault();
		console.log(formLoginValues);
	};

	return (
		<div className="container login-container">
			<div className="row">
				<div className="col-md-6 login-form-1">
					<h3>Ingreso</h3>

					<form onSubmit={handleLogin}>
						<div className="form-group">
							<input
								type="text"
								className="form-control"
								placeholder="email"
								name="loginEmail"
								value={loginEmail}
								onChange={handleLoginInputChange}
							/>
						</div>
						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="password"
								name="loginPassword"
								value={loginPassword}
								onChange={handleLoginInputChange}
							/>
						</div>
						<div className="form-group">
							<input type="submit" className="btnSubmit" value="Login" />
						</div>
					</form>
				</div>

				<div className="col-md-6 login-form-2">
					<h3>Registro</h3>
					<form>
						<div className="form-group">
							<input
								type="text"
								className="form-control"
								placeholder="Nombre"
							/>
						</div>
						<div className="form-group">
							<input
								type="email"
								className="form-control"
								placeholder="Correo"
							/>
						</div>
						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="Contraseña"
							/>
						</div>

						<div className="form-group">
							<input
								type="password"
								className="form-control"
								placeholder="Repita la contraseña"
							/>
						</div>

						<div className="form-group">
							<input type="submit" className="btnSubmit" value="Crear cuenta" />
						</div>
					</form>
				</div>
			</div>
		</div>
	);
};

export default LoginScreen;
