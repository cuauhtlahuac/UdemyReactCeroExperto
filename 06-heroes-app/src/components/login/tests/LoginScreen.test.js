import { AuthContext } from 'contexts/authContext';
import React, { PureComponent } from 'react';
import { types } from 'types/types';
import LoginScreen from '../LoginScreen';
const { shallow, mount } = require('enzyme');

describe('<Login Screen tests /> Tests', () => {
  const history = {
    replace: jest.fn(),
  }

	const context = {
		dispatch: jest.fn(),
	};

	const component = mount(
		<AuthContext.Provider value={context}>
			<LoginScreen history={history} />
		</AuthContext.Provider>,
	);

	test('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});

	test('should render correctly', () => {
		expect(component.find('div.m-5 h1').text()).toBe('Login');
	});

	test('should call dispatch', () => {
		component.find('button').prop('onClick')();

		expect(context.dispatch).toHaveBeenCalledWith({
			type: types.login,
			payload: { name: 'Cuauhtlahuac' },
		});
	});
});
