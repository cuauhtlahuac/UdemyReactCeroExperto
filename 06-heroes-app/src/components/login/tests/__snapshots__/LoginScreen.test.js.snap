// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login Screen tests /> Tests should render correctly 1`] = `
<div
  className="m-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
